src/k8s/pkg/k8sd/api/cluster_tokens.go

```diff
@@ -54,9 +54,11 @@ func getOrCreateJoinToken(ctx context.Context, m *microcluster.MicroCluster, tok
 		fmt.Println("No token exists yet. Creating a new token.")
 	}
 
-	// if token does not exist, create a new one
-	// TODO(ben): make token expiry configurable
-	token, err := m.NewJoinToken(ctx, tokenName, 24*time.Hour)
+	if ttl == 0 {
```

Comment:
This feels like we're having 2 defaults being set in 2 different places. Maybe we can throw an error here since we know `req.TTL` should be set (otherwise it's an error on the k8sd caller side that we need to fix)? After all k8s snap and k8sd are shipped together, we're not doing this for backwards compatibility, right?

Reply:
Good point, but the CLI is not the only way you can configure/control `k8sd`. For example, the `k8s-operator` charm (https://github.com/canonical/k8s-operator/blob/113b69cc428aaf7e1ad338c5b71b660df3e062c3/charms/worker/k8s/lib/charms/k8s/v0/k8sd_api_manager.py#L171) controls the snap directly over the API.
So, we need to provide a default for those cases.

You could argue that we don't need to set the default in the CLI. However, I'd say that people should not need to go all the way to the back-end implementation to find out the default for this value. Since it would be fine if backend default and CLI default would diverge, I guess it is fine to set this twice.

Reply:
Oh, I didn't know about the operator. Thanks for the context.